pennylane_lightning/src/Gates.cpp

```diff
@@ -407,6 +447,19 @@ void Pennylane::CRotationXGate::applyKernel(const StateVector& state, const std:
     }
 }
 
+const double Pennylane::CRotationXGate::generatorScalingFactor{ -0.5 };
+
+void Pennylane::CRotationXGate::applyGenerator(const StateVector& state, const std::vector<size_t>& indices, const std::vector<size_t>& externalIndices) {
+    for (const size_t& externalIndex : externalIndices) {
+        CplxType* shiftedState = state.arr + externalIndex;
+        CplxType v0 = shiftedState[indices[2]];
+        CplxType v1 = shiftedState[indices[3]];
+        shiftedState[indices[0]] = shiftedState[indices[1]] = 0;
+        shiftedState[indices[2]] = js * v0 + c * v1;
+        shiftedState[indices[3]] = c * v0 + js * v1;
+    }
+}
+
```

Comment:
I'm not sure about the generators for the following 3 gates. For me, the generator for this one would be the CNOT gate. The generator for `CRotationYGate` would be a controlled version of the `PauliY` kernel - something like:
```python
void Pennylane::CRotationXGate::applyGenerator(const StateVector& state, const std::vector<size_t>& indices, const std::vector<size_t>& externalIndices, bool) {
    for (const size_t& externalIndex : externalIndices) {
        CplxType* shiftedState = state.arr + externalIndex;
        CplxType v2 = shiftedState[indices[2]];
        shiftedState[indices[2]] = -IMAG * shiftedState[indices[3]];
        shiftedState[indices[3]] = IMAG * v2;
    }
}
```

Reply:
Oh, yep, these are wrong. Thanks for catching it! I managed to confuse applying the derivative with applying the generator here. What I wrote should (I think) apply the derivative and _not_ the generator. I just copied them a bit too quickly. 😆